src/execution_plan/optimizations/traverse_order_utils.c

```diff
@@ -199,10 +199,10 @@ void FilterTree_CollectIndependentEntities
 		}
 
 		raxFree(modified);
-		FilterTree_Free(t);
 	}
 
 	array_free(sub_trees);
+	FilterTree_Free(tree);
```

Comment:
in this case I don't see a reason to call `FT_FilterNode *tree = FilterTree_Clone(root);` in the first place, this can save the call to `FilterTree_Free(tree);`